CMakeLists.txt

```diff
@@ -22,6 +22,7 @@ set(PACKAGE_BUGREPORT "https://github.com/googleapis/google-cloud-cpp/issues")
 project(
     google-cloud-cpp
     VERSION 2.5.0
+    DESCRIPTION "rc"  # pre-release version
```

Comment:
That is a bit cheeky.  `DESCRIPTION` should be used for something like "The Google Cloud C++ SDK".

There is a long discussion of a related topic here:

https://gitlab.kitware.com/cmake/cmake/-/issues/16716

It includes some alternatives that may work for us, e.g.:

```CMake
project(google-cloud-cpp VERSION 2.5.0 LANGUAGES CXX)
set(GOOGLE_CLOUD_CPP_PRE_RELEASE "-rc")
set(PROJECT_VERSION "${PROJECT_VERSION}${GOOGLE_CLOUD_CPP_PRE_RELEASE")
```

and then in `version_info.h.in`

```cc
#define GOOGLE_CLOUD_CPP_PRE_RELEASE "@GOOGLE_CLOUD_CPP_PRE_RELEASE@"
```

Reply:
Yes, a little cheeky.  Anyway, take a look at new impl along the lines suggested.

Reply:
SGTM